internal/connect/utils.go

```diff
@@ -10,15 +10,15 @@ import (
 	"github.com/confluentinc/cli/v3/pkg/errors"
 )
 
-func getConfig(cmd *cobra.Command) (*map[string]string, error) {
+func getConfigAndOffsets(cmd *cobra.Command, isUpdate bool) (*map[string]string, *[]map[string]any, error) {
```

Comment:
Maps and slices are reference types, so they are always passed by reference. You don't need a pointer.
```suggestion
func getConfigAndOffsets(cmd *cobra.Command, isUpdate bool) (map[string]string, []map[string]any, error) {
```